### PR TITLE
Add checklist_updated to track when a teacher has edited the checlist

### PR DESCRIPTION
**Pull request by andrewhancox**

### CODE REVIEW

File: locallib.php [hunk 1]
```diff
@@ -1863,50 +1867,71 @@ protected function process_edit_actions() {
...
 
             default:
                 throw new moodle_exception('invalidaction', 'mod_checklist', '', $action);
         }
 
+        if ($triggerupdate) {
+            $params = array(
+                'contextid' => $this->context->id,
+                'objectid' => $this->checklist->id,
+                'userid' => $USER->id,
```
Comment (davosmith) [inline locallib.php:1929]: userid should be filled in automatically from $USER->id by ::create(), if none is provided? (I've not double-checked)

File: classes/event/checklist_updated.php [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+class checklist_updated extends \core\event\base {
+
+    /**
+     * Init method.
+     *
+     * @return void
+     */
+    protected function init() {
+        $this->data['crud'] = 'u';
+        $this->data['edulevel'] = self::LEVEL_PARTICIPATING;
```
Comment (davosmith) [inline classes/event/checklist_updated.php:44]: Maybe LEVEL_TEACHING ? (I'm not really sure - and it probably doesn't matter)

File: version.php [hunk 3]
```diff
@@ -23,7 +23,7 @@
 
 defined('MOODLE_INTERNAL') || die();
 
-$plugin->version = 2024020200;  // The current module version (Date: YYYYMMDDXX).
+$plugin->version = 2024020201;  // The current module version (Date: YYYYMMDDXX).
```
Comment (davosmith) [inline version.php:26]: Possibly bump it all the way up to 2024032200

**Comment by andrewhancox** [comment]
Crikey Davo, that was quick even for you!
All good points, I did this in a massive rush.
Hope all is good with you and yours!

**Comment by davosmith** [comment]
I saw your name on the incoming email and was curious to see what it was that you'd changed.

Looks to be sensible code, so I'll merge it, assuming it passes the automated tests.

I'll need to update the README, but, apart from that, I should be able to release it fairly quickly (as it's a small, uncontroversial change).

**Comment by davosmith** [comment]
That all looks to have passed automated tests.

I'll merge for now and sort out README + release later.

**Comment by andrewhancox** [comment]
Thanks!

**Pull request merged**